python/observability-online-eval/coding-agent/coding_agent/execution.py

```
@@ -0,0 +1,72 @@
+            try:
+                filepath.relative_to(workdir_path)
+            except ValueError as exc:
+                raise ValueError(f"Invalid path outside workdir: {filename}") from exc
+            filepath.parent.mkdir(parents=True, exist_ok=True)
+            with filepath.open("w") as f:
+                f.write(content)
+
+    def execute(self, command: str = "bash run.sh", timeout: int = 30) -> ExecutionResult:
+        """Run command via subprocess, capture stdout/stderr."""
+        try:
+            result = subprocess.run(
```

Ruff (0.15.4)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)